hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileStatus.java

```diff
@@ -54,12 +57,30 @@ public S3AFileStatus(boolean isemptydir,
   public S3AFileStatus(Tristate isemptydir,
       Path path,
       String owner) {
-    super(0, true, 1, 0, 0, 0,
-        null, null, null, null,
+    this(isemptydir, path, owner, owner, 0, 0, null);
+  }
+
+  /**
+   * Create a directory status.
+   * @param isemptydir is this an empty directory?
+   * @param path the path
+   * @param owner the owner
+   * @param group the group
+   * @param modification_time the modification time
+   * @param access_time the access time
+   * @param permission the permission
+   */
+  public S3AFileStatus(Tristate isemptydir,
```

Comment:
what's the reason for adding permissions here? There aren't any with S3 and so the default of rwx is thrown up, for better or (sometimes) worse. The risk is that if some restricted permissions are set then code may actually believe those permissions are enforced, when they aren't. Same for (user, group)

Reply:
Good point.  I made the change after observing test failures.  Dropping those attributes back out of here I see it looks like TestLocalMetadataStore is one test that fails.  For example, LocalMetadataStore.testPutOverwrite() fails in S3ATestUtils.verifyFileStatus().  MetaStoreTestBase.basicFileStatus() creates a FileStatus with permission and group set.  TestLocalMetadataStore.verifyFileStatus() verifies these attributes are preserved.

It sounds like I should probably change those tests such that they don't use or expect these unsupported attributes?

The test previously worked because PathMetadata carried a vanilla FileStatus.  I converted it over to carrying S3AFileStatus, which without these changes doesn't allow the group and permissions attributes forward.

Reply:
> It sounds like I should probably change those tests such that they don't use or expect these unsupported attributes?

Yes I think so

Reply:
Addressed in https://github.com/apache/hadoop/pull/675/commits/bee4e5291f216bc3e35ca7765873dcd8d6355648